src/shared_modules/content_manager/README.md

```diff
@@ -0,0 +1,341 @@
+Output folders created.
+FactoryContentUpdater - Starting process
+Creating 'cti-api' downloader
+Creating 'raw' content decompressor
+Creating 'cti-api' version updater
+Downloaded content cleaner created
+FactoryContentUpdater - Finishing process
+ActionOrchestrator - Finishing process
+ActionOrchestrator - Running process
+CtiApiDownloader - Starting
+CtiApiDownloader - Request processed successfully.
+CtiApiDownloader - Finishing
+SkipStep - Executing
+PubSubPublisher - No data to publish
+All files in the folder have been deleted.
+```
+
+### Use case: Download from regular API
+
+The Content Manager can also download content from a regular API. The functionality is quite straightforward: Download from a given URL and store the content in an output file. No API parameters are handled nor added by the module.
+
+```json
+{
+    "topicName": "API content download",
+    "interval": 5,
+    "ondemand": false,
+    "configData":
+    {
+        "contentSource": "api",
+        "compressionType": "raw",
+        "url": "https://jsonplaceholder.typicode.com/todos/1",
+        "outputFolder": "/tmp/output_folder",
+        "contentFileName": "content.json"
+    }
+}
+```
+
+> For simplicity, only the usage case-related configurations are shown.
+
+The configuration above will make the Content Manager launch each `5` seconds an orchestration that will download the content from the API `https://jsonplaceholder.typicode.com/todos/1` and store the content in `/tmp/output_folder`.
+
+
+```bash
+# ./content_manager_test_tool
+ActionOrchestrator - Starting process
+API offset to be used: 0
+Output folders created.
+FactoryContentUpdater - Starting process
+Creating 'api' downloader
+Creating 'raw' content decompressor
+Version updater not needed
+Downloaded content cleaner created
+FactoryContentUpdater - Finishing process
+ActionOrchestrator - Finishing process
+ActionOrchestrator - Running process
+APIDownloader - Starting
+APIDownloader - Finishing - Download done successfully
+SkipStep - Executing
+PubSubPublisher - Data published
+SkipStep - Executing
+All files in the folder have been deleted.
+```
+
+The downloaded content will be all stored in a unique file called `content.json`:
+
+```bash
+# tree /tmp/output_folder/  
+/tmp/output_folder/
+|-- contents
+|   `-- content.json
+`-- downloads
+
+2 directories, 1 file
+```
+
+```bash
+# cat /tmp/output_folder/contents/content.json
+{
+  "userId": 1,
+  "id": 1,
+  "title": "delectus aut autem",
+  "completed": false
+}
+```
+
+> If the content was compressed, the output file would be stored in the _downloads_ folder.
+
+The `/tmp/output_folder/contents/content.json` path is published for the consumers to read.
+
+### Use case: Remote file download
+
+The Content Manager has the capability of downloading a content file from a URL. The file will be stored in the output folder and, if compressed, it will be decompressed in the contents folder.
```

Comment:
Improve (clarity): I think this is a bit hard to follow, maybe show the tree structure along side (also applies to the other examples)

Reply:
I'm already providing the tree structure of the output in every example. You have to scroll down a little bit though.